test/gas/EndToEnd.t.sol

```diff
@@ -0,0 +1,103 @@
+import {SmartWalletTestBase} from "../CoinbaseSmartWallet/SmartWalletTestBase.sol";
+import {CoinbaseSmartWallet} from "../../src/CoinbaseSmartWallet.sol";
+import {CoinbaseSmartWalletFactory} from "../../src/CoinbaseSmartWalletFactory.sol";
+import {MockTarget} from "../mocks/MockTarget.sol";
+import {UserOperation} from "account-abstraction/interfaces/UserOperation.sol";
+import {MockERC20} from "../../lib/solady/test/utils/mocks/MockERC20.sol";
+import {Static} from "../CoinbaseSmartWallet/Static.sol";
+
+/// forge-config: default.isolate = true
+contract EndToEndTest is SmartWalletTestBase {
+    address eoaUser = address(0xe0a);
+    
+    MockERC20 usdc;
+    MockTarget target;
+    CoinbaseSmartWalletFactory factory;
+    
+    function setUp() public override {
+        vm.etch(0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789, Static.ENTRY_POINT_BYTES);
+        
+        CoinbaseSmartWallet implementation = new CoinbaseSmartWallet();
+        factory = new CoinbaseSmartWalletFactory(address(implementation));
+        
+        signerPrivateKey = 0xa11ce;
+        signer = vm.addr(signerPrivateKey);
+        owners.push(abi.encode(signer));
+        account = factory.createAccount(owners, 0);
+        
+        vm.deal(address(account), 100 ether);
+        vm.deal(eoaUser, 100 ether);
+        
+        usdc = new MockERC20("USD Coin", "USDC", 6);
+        usdc.mint(address(account), 10000e6);
+        usdc.mint(eoaUser, 10000e6);
+        
+        target = new MockTarget();
+    }
+    
+    // Native ETH transfer comparison between ERC-4337 and EOA
+    function test_transfer_native() public {
+        userOpCalldata = abi.encodeCall(CoinbaseSmartWallet.execute, (address(0x1234), 1 ether, ""));
+        UserOperation memory op = _getUserOpWithSignature();
+        
+        bytes memory handleOpsCalldata = abi.encodeCall(entryPoint.handleOps, (_makeOpsArray(op), payable(bundler)));
+        console2.log("test_transfer_native ERC-4337 calldata size:", handleOpsCalldata.length);
+        
+        vm.startSnapshotGas("e2e_transfer_native_4337");
+        _sendUserOperation(op);
+        uint256 gas4337 = vm.stopSnapshotGas();
+        console2.log("test_transfer_native ERC-4337 gas:", gas4337);
+        
+        console2.log("test_transfer_native EOA calldata size:", uint256(0));
+        
+        vm.prank(eoaUser);
+        vm.startSnapshotGas("e2e_transfer_native_eoa");
+        payable(address(0x1234)).transfer(1 ether);
+        uint256 gasEOA = vm.stopSnapshotGas();
+        console2.log("test_transfer_native EOA gas:", gasEOA);
+        console2.log("Gas overhead (4337/EOA):", (gas4337 * 100) / gasEOA, "%");
+    }
+    
+    // ERC20 transfer comparison between ERC-4337 and EOA
+    function test_transfer_erc20() public {
+        userOpCalldata = abi.encodeCall(
+            CoinbaseSmartWallet.execute,
+            (address(usdc), 0, abi.encodeCall(usdc.transfer, (address(0x5678), 100e6)))
+        );
```

Comment:
same thing here as previous function with:
1. fuzzing instead of hardcoding values
2. dust recipients to remove variability from zero->nonzero storage transition